apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,125 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile.
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	// +optional
+	DisplayName string `json:"displayName"`
+
+	// ClusterManager defines which cluster manager owns this ClusterProfile resource
+	// +required
+	ClusterManager ClusterManager `json:"clusterManager"`
+}
+
+// ClusterManager defines which cluster manager owns this ClusterProfile resource.
+// This field is immutable.
+// It's recommended that each cluster manager instance should set a different values to this field.
+// In addition, it's recommended that a predefined label with key "x-k8s.io/cluster-manager"
+// should be added by the cluster manager upon creation. See constant LabelClusterManager.
+// The value of the label should be the same as the name of the cluster manager.
+// The purpose of this label is to make filter clusters from different cluster managers easier.
+//
+// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ClusterManager is immutable"
+type ClusterManager struct {
+	// Name defines the name of the cluster manager
+	// +required
+	Name string `json:"name"`
+}
+
+// ClusterProfileStatus defines the observed state of ClusterProfile.
+type ClusterProfileStatus struct {
+	// Conditions contains the different condition statuses for this cluster.
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Version defines the version information of the cluster.
+	Version ClusterVersion `json:"version,omitempty"`
+
+	// Properties defines properties of collected from the cluster.
+	// For example, a unique cluster identifier (id.k8s.io).
+	// The set of properties is not uniform across a fleet, some properties can be
+	// vendor or version specific and may not be included from all clusters.
+	// +optional
+	Properties []Property `json:"properties,omitempty"`
+}
+
+// ClusterVersion represents version information about the cluster.
+type ClusterVersion struct {
+	// Kubernetes is the kubernetes version of the cluster.
+	// +optional
+	Kubernetes string `json:"kubernetes,omitempty"`
+}
+
+// Property defines a property collected from a cluster.
+type Property struct {
+	// Name is the name of a property resource on cluster. It's a well-known
+	// or customized name to identify the propertie.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// Value is a property-dependent string
+	// +kubebuilder:validation:MaxLength=1024
+	// +kubebuilder:validation:MinLength=1
+	Value string `json:"value"`
+}
+
+const (
+	// ClusterConditionJoined means the cluster has successfully joined the control.
+	ClusterConditionJoined string = "Joined"
+	// ClusterConditionHealthy means the cluster is healthy.
+	ClusterConditionHealthy string = "Healthy"
```

Comment:
I've removed the Joined condition for now to avoid the confusion. See https://github.com/kubernetes-sigs/cluster-inventory-api/pull/6#discussion_r1572638760 for more details.

Reply:
This is less confusing. We may want to add other conditions for "unhealthy" cases, for example, we may want to have a "disconnected" status for a cluster that is registered to a fleet, but it cannot be reached.

Reply:
How about "available" and "unavailable" instead of healthy/unhealthy/connected/disconnected? 

In the SIG MC meetings and the KEP PR, we had multiple lengthy discussions what healthy and unhealthy means. I feel like available and unavailable is more generic for the consumers instead of diving deep into what healthy and connection status means for now.

Reply:
'available/unavailable' sounds good to me.

Reply:
After thinking more about this, I am still not clear what it signifies. I assume, there will be a controller somewhere that updates this status. The controller may be running on the manager or on the member clusters. When we set the status to "available" does it mean that the control plane of the cluster is available? Or, does it mean that the data plane of the cluster is "available"?
Practical scenarios:
1. The controller which sets this status runs on the control plane of the cluster and sends us a heartbeat. It makes the cluster look available while its data plane might be down.
2. The controller runs on a node in the member cluster and sends us a heartbeat. The cluster will look available even when its control plane is down.
3. The controller runs on the manager and checks a "ping" endpoint, depending on where that endpoint runs, we will have one of the above scenarios.

An alternative to the above is to have more logic that checks both control plane and data plane and marks the cluster available only when both are functional.
This long comment is here to just say that we should define these conditions more precisely.

Reply:
Reworked this to be `ControlPlaneHealthy`. PTAL. 

The data plane healthiness condition is a lot more difficult to define. It will probably require it's own separate PR. CC @ArangoGutierrez

Reply:
Thank you. It is clear now. "Conditions" is a required field of ClusterProfileStatus and we currently have only this predefined condition. Do we expect this condition to be always present in ClusterProfileStatus?

Reply:
Yes, I think it's important for the consumers to be able to rely on the existence of the conditions field. If the cluster manager is unable to determine the `ControlPlaneHealthy`, setting the status to `Unknown` is acceptable.

Reply:
Spec and status are updated independently.  Having `Conditions` be required means there MUST BE a condition specified when the ClusterProfile is created.  This is not something I have ever seen before, and I would argue that if you have a required condition, you should just use a proper field.

Reply:
You are right. Based on the core types https://github.com/kubernetes/api/blob/master/core/v1/types.go all the condition fields are optional. I am going to fix this. 
My original thinking was Status is marked optional so if a status is populated then a required condition inside the status seems reasonable.